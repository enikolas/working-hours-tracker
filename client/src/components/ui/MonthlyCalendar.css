.react-datepicker {
  border: 0;
  background-color: transparent;
  width: calc(100% - 15px);
  margin: 30px 0 30px 15px;
}

.react-datepicker__month-container {
  width: 100%;
}

.react-datepicker__header {
  width: 100%;
  background: transparent;
  border: 0;
  border-bottom: 1px solid #5a717f;
  border-radius: 0;
}

.react-datepicker__current-month {
  color: #254356;
  text-transform: uppercase;
}

.react-datepicker__day {
  color: #fff;
  transition: background 0.3s, color 0.3s;
  border-radius: 0;
}

.react-datepicker__day:hover {
  border-radius: 0;
  background: rgba(255,255,255,0.2);
}

.react-datepicker__day--weekend {
  opacity: 0.6;
}

.react-datepicker__day--disabled {
  color: #c5ccd1;
}

.react-datepicker__day.active {
  color: #254356;
}

.react-datepicker__day--selected {
  outline: none;
  background: #5a717f;
  color: #f93;
}

.react-datepicker__day--selected:hover {
  background: #c5ccd1;
}

.react-datepicker__day--outside-month {
  opacity: 0.5;
}

.react-datepicker__day-name {
  color: #c5ccd1;
}

.react-datepicker__navigation--next {
  border-left-color: #fff;
}

.react-datepicker__navigation--previous {
  border-right-color: #fff;
}

.calendar-checked,
.calendar-unchecked,
.calendar-vacation,
.calendar-holiday {
  position: relative;
}

.calendar-checked:after,
.calendar-unchecked:after,
.calendar-vacation:after,
.calendar-holiday:after {
  content: 'âœ“';
  position: absolute;
  top: 10px;
  color: #479030;
}

.calendar-unchecked:after {
  content: 'âœ—';
  color: #dc3545;
}

.calendar-vacation:after {
  content: 'ðŸŒ´';
  color: #bf0;
}

.calendar-holiday:after {
  content: 'âœª';
  color: #ff0;
}

.calendar-absence:after {
  content: 'ðŸ“„';
  color: #fff;
}

.calendar-otanjoubi:after {
  content: 'ðŸŽ‰';
  color: #f99;
}
