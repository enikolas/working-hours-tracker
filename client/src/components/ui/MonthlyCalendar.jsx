import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment-timezone';
import DatePicker from 'react-datepicker';

import 'react-datepicker/dist/react-datepicker.css';

import './MonthlyCalendar.css';

const _getStyleClassForCalendarDays = (timeEntries) => {
  const checked = [];
  const unchecked = [];
  const futureDay = [];

  const holidays = [];
  const vacations = [];
  const otanjoubis = [];
  const absences = [];

  timeEntries.forEach((dayEntry) => {
    const day = moment(dayEntry.date).startOf('day').toDate();

    if (dayEntry.isHoliday) {
      holidays.push(day);
    } else if (dayEntry.isVacation) {
      vacations.push(day);
    } else if (dayEntry.isOtanjoubi) {
      otanjoubis.push(day);
    } else if (dayEntry.isJustifiedAbsence) {
      absences.push(day);
    } else if (dayEntry.total && dayEntry.total !== '0:00') {
      checked.push(day);
    } else {
      unchecked.push(day);
    }

    const isDayAfterToday = moment(day).isAfter(moment(), 'day');
    if (isDayAfterToday) {
      futureDay.push(day);
    }
  });

  return [
    { 'calendar-checked': checked },
    { 'calendar-unchecked': unchecked },
    { 'calendar-holiday': holidays },
    { 'calendar-vacation': vacations },
    { 'calendar-otanjoubi': otanjoubis },
    { 'calendar-absence': absences },
    { 'calendar-future-day': futureDay }
  ];
};

const MonthlyCalendar = ({ selectedDate, onDateChange, timeEntries }) => {
  return selectedDate && (
    <DatePicker
      inline
      highlightDates={_getStyleClassForCalendarDays(timeEntries)}
      selected={selectedDate.toDate()}
      onChange={onDateChange}
      filterDate={date => moment(date).isSameOrBefore(moment(), 'day')}
      maxTime={new Date()}
    />
  );
};

MonthlyCalendar.propTypes = {
  selectedDate: PropTypes.object,
  onDateChange: PropTypes.func,
  timeEntries: PropTypes.array
};

MonthlyCalendar.defaultProps = {
  selectedDate: {},
  onDateChange: () => {},
  timeEntries: []
};

export default MonthlyCalendar;
