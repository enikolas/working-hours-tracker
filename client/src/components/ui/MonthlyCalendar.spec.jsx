import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment-timezone';

import MonthlyCalendar from './MonthlyCalendar';

const defaultProps = {
	selectedDate: moment('2018-02-14T00:00:00.000Z'),
	onDateChange: jest.fn(),
	timeEntries: [
		{
			date: '2018-02-11',
			phase: 'Project Phase Name',
			activity: 'Code/ Document Reviews',
			startTime: '9:28',
			startBreakTime: '11:35',
			endBreakTime: '12:35',
			endTime: '18:28',
			total: '8:00',
			__typename: 'TimeEntry'
		},
		{
			date: '2018-02-12',
			phase: 'Project Phase Name',
			activity: 'Code/ Document Reviews',
			startTime: '9:28',
			startBreakTime: '11:35',
			endBreakTime: '12:35',
			endTime: '18:28',
			total: '8:00',
			__typename: 'TimeEntry'
		},
		{
			date: '2018-02-13',
			phase: 'Project Phase Name',
			activity: 'Code/ Document Reviews',
			startTime: '9:28',
			startBreakTime: '11:35',
			endBreakTime: '12:35',
			endTime: '18:28',
			total: '8:00',
			__typename: 'TimeEntry'
		},
		{
			date: '2018-02-14',
			phase: '',
			activity: '',
			startTime: '',
			startBreakTime: '',
			endBreakTime: '',
			endTime: '',
			total: '',
			__typename: 'TimeEntry'
		},
		{
			date: '2018-02-15',
			phase: 'Project Phase Name',
			activity: 'Code/ Document Reviews',
			startTime: '9:28',
			startBreakTime: '11:35',
			endBreakTime: '12:35',
			endTime: '18:28',
			total: '8:00',
			__typename: 'TimeEntry'
		},
		{
			date: '2018-02-16',
			phase: 'Project Phase Name',
			activity: 'Code/ Document Reviews',
			startTime: '9:28',
			startBreakTime: '',
			endBreakTime: '',
			endTime: '18:28',
			total: '8:00',
			__typename: 'TimeEntry'
		},
		{
			date: '2018-02-17',
			phase: 'Project Phase Name',
			activity: 'Code/ Document Reviews',
			startTime: '9:28',
			startBreakTime: '11:35',
			endBreakTime: '12:35',
			endTime: '18:28',
			total: '8:00',
			__typename: 'TimeEntry'
		}
	]
};

describe('MonthlyCalendar', () => {
	Date.now = jest.fn(() => new Date(Date.UTC(2018, 1, 14, 12)).valueOf());

	describe('render', () => {
		it('should render DatePicker with style', () => {
			Date.now = jest.fn(() => new Date(Date.UTC(2018, 1, 14, 12)).valueOf());

			const wrapper = shallow(<MonthlyCalendar
				selectedDate={defaultProps.selectedDate}
				onDateChange={defaultProps.onDateChange}
				timeEntries={defaultProps.timeEntries}
			/>);

			expect(wrapper).toMatchSnapshot();
		});

		it('should show if a day in past is blocked', () => {
			Date.now = jest.fn(() => new Date(Date.UTC(2018, 1, 22, 12)).valueOf());

			const nextWeekWrapper = shallow(<MonthlyCalendar
				selectedDate={moment('2018-02-21T00:00:00.000Z')}
				onDateChange={defaultProps.onDateChange}
				timeEntries={defaultProps.timeEntries}
			/>);

			expect(nextWeekWrapper).toMatchSnapshot();
		});

		it('should render without timeEntries', () => {
			const wrapper = shallow(<MonthlyCalendar
				selectedDate={defaultProps.selectedDate}
				onDateChange={defaultProps.onDateChange}
			/>);

			expect(wrapper).toMatchSnapshot();
		});
	});
});
