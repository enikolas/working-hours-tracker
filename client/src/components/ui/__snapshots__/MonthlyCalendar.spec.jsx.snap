// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthlyCalendar render should render DatePicker with style 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MonthlyCalendar
    onDateChange={[MockFunction]}
    selectedDate={"2018-02-14T00:00:00.000Z"}
    timeEntries={
      Array [
        Object {
          "__typename": "TimeEntry",
          "activity": "Code/ Document Reviews",
          "date": "2018-02-11",
          "endBreakTime": "12:35",
          "endTime": "18:28",
          "phase": "Project Phase Name",
          "startBreakTime": "11:35",
          "startTime": "9:28",
          "total": "8:00",
        },
        Object {
          "__typename": "TimeEntry",
          "activity": "Code/ Document Reviews",
          "date": "2018-02-12",
          "endBreakTime": "12:35",
          "endTime": "18:28",
          "phase": "Project Phase Name",
          "startBreakTime": "11:35",
          "startTime": "9:28",
          "total": "8:00",
        },
        Object {
          "__typename": "TimeEntry",
          "activity": "Code/ Document Reviews",
          "date": "2018-02-13",
          "endBreakTime": "12:35",
          "endTime": "18:28",
          "phase": "Project Phase Name",
          "startBreakTime": "11:35",
          "startTime": "9:28",
          "total": "8:00",
        },
        Object {
          "__typename": "TimeEntry",
          "activity": "",
          "date": "2018-02-14",
          "endBreakTime": "",
          "endTime": "",
          "phase": "",
          "startBreakTime": "",
          "startTime": "",
          "total": "",
        },
        Object {
          "__typename": "TimeEntry",
          "activity": "Code/ Document Reviews",
          "date": "2018-02-15",
          "endBreakTime": "12:35",
          "endTime": "18:28",
          "phase": "Project Phase Name",
          "startBreakTime": "11:35",
          "startTime": "9:28",
          "total": "8:00",
        },
        Object {
          "__typename": "TimeEntry",
          "activity": "Code/ Document Reviews",
          "date": "2018-02-16",
          "endBreakTime": "",
          "endTime": "18:28",
          "phase": "Project Phase Name",
          "startBreakTime": "",
          "startTime": "9:28",
          "total": "8:00",
        },
        Object {
          "__typename": "TimeEntry",
          "activity": "Code/ Document Reviews",
          "date": "2018-02-17",
          "endBreakTime": "12:35",
          "endTime": "18:28",
          "phase": "Project Phase Name",
          "startBreakTime": "11:35",
          "startTime": "9:28",
          "total": "8:00",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "allowSameDay": false,
      "dateFormat": "L",
      "dateFormatCalendar": "MMMM YYYY",
      "disabled": false,
      "disabledKeyboardNavigation": false,
      "dropdownMode": "scroll",
      "filterDate": [Function],
      "highlightDates": Array [
        Object {
          "calendar-checked": Array [
            "2018-02-11T02:00:00.000Z",
            "2018-02-12T02:00:00.000Z",
            "2018-02-13T02:00:00.000Z",
            "2018-02-15T02:00:00.000Z",
            "2018-02-16T02:00:00.000Z",
            "2018-02-17T02:00:00.000Z",
          ],
        },
        Object {
          "calendar-unchecked": Array [
            "2018-02-14T02:00:00.000Z",
          ],
        },
        Object {
          "calendar-holiday": Array [],
        },
        Object {
          "calendar-vacation": Array [],
        },
        Object {
          "calendar-otanjoubi": Array [],
        },
        Object {
          "calendar-absence": Array [],
        },
        Object {
          "calendar-future-day": Array [
            "2018-02-15T02:00:00.000Z",
            "2018-02-16T02:00:00.000Z",
            "2018-02-17T02:00:00.000Z",
          ],
        },
      ],
      "inline": true,
      "maxTime": "2018-02-14T12:00:00.000Z",
      "monthsShown": 1,
      "onBlur": [Function],
      "onChange": [MockFunction],
      "onClickOutside": [Function],
      "onFocus": [Function],
      "onKeyDown": [Function],
      "onMonthChange": [Function],
      "onSelect": [Function],
      "onYearChange": [Function],
      "preventOpenOnFocus": false,
      "selected": "2018-02-14T00:00:00.000Z",
      "shouldCloseOnSelect": true,
      "showTimeSelect": false,
      "timeCaption": "Time",
      "timeIntervals": 30,
      "withPortal": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowSameDay": false,
        "dateFormat": "L",
        "dateFormatCalendar": "MMMM YYYY",
        "disabled": false,
        "disabledKeyboardNavigation": false,
        "dropdownMode": "scroll",
        "filterDate": [Function],
        "highlightDates": Array [
          Object {
            "calendar-checked": Array [
              "2018-02-11T02:00:00.000Z",
              "2018-02-12T02:00:00.000Z",
              "2018-02-13T02:00:00.000Z",
              "2018-02-15T02:00:00.000Z",
              "2018-02-16T02:00:00.000Z",
              "2018-02-17T02:00:00.000Z",
            ],
          },
          Object {
            "calendar-unchecked": Array [
              "2018-02-14T02:00:00.000Z",
            ],
          },
          Object {
            "calendar-holiday": Array [],
          },
          Object {
            "calendar-vacation": Array [],
          },
          Object {
            "calendar-otanjoubi": Array [],
          },
          Object {
            "calendar-absence": Array [],
          },
          Object {
            "calendar-future-day": Array [
              "2018-02-15T02:00:00.000Z",
              "2018-02-16T02:00:00.000Z",
              "2018-02-17T02:00:00.000Z",
            ],
          },
        ],
        "inline": true,
        "maxTime": "2018-02-14T12:00:00.000Z",
        "monthsShown": 1,
        "onBlur": [Function],
        "onChange": [MockFunction],
        "onClickOutside": [Function],
        "onFocus": [Function],
        "onKeyDown": [Function],
        "onMonthChange": [Function],
        "onSelect": [Function],
        "onYearChange": [Function],
        "preventOpenOnFocus": false,
        "selected": "2018-02-14T00:00:00.000Z",
        "shouldCloseOnSelect": true,
        "showTimeSelect": false,
        "timeCaption": "Time",
        "timeIntervals": 30,
        "withPortal": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MonthlyCalendar render should render without timeEntries 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MonthlyCalendar
    onDateChange={[MockFunction]}
    selectedDate={"2018-02-14T00:00:00.000Z"}
    timeEntries={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "allowSameDay": false,
      "dateFormat": "L",
      "dateFormatCalendar": "MMMM YYYY",
      "disabled": false,
      "disabledKeyboardNavigation": false,
      "dropdownMode": "scroll",
      "filterDate": [Function],
      "highlightDates": Array [
        Object {
          "calendar-checked": Array [],
        },
        Object {
          "calendar-unchecked": Array [],
        },
        Object {
          "calendar-holiday": Array [],
        },
        Object {
          "calendar-vacation": Array [],
        },
        Object {
          "calendar-otanjoubi": Array [],
        },
        Object {
          "calendar-absence": Array [],
        },
        Object {
          "calendar-future-day": Array [],
        },
      ],
      "inline": true,
      "maxTime": "2018-02-22T12:00:00.000Z",
      "monthsShown": 1,
      "onBlur": [Function],
      "onChange": [MockFunction],
      "onClickOutside": [Function],
      "onFocus": [Function],
      "onKeyDown": [Function],
      "onMonthChange": [Function],
      "onSelect": [Function],
      "onYearChange": [Function],
      "preventOpenOnFocus": false,
      "selected": "2018-02-14T00:00:00.000Z",
      "shouldCloseOnSelect": true,
      "showTimeSelect": false,
      "timeCaption": "Time",
      "timeIntervals": 30,
      "withPortal": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowSameDay": false,
        "dateFormat": "L",
        "dateFormatCalendar": "MMMM YYYY",
        "disabled": false,
        "disabledKeyboardNavigation": false,
        "dropdownMode": "scroll",
        "filterDate": [Function],
        "highlightDates": Array [
          Object {
            "calendar-checked": Array [],
          },
          Object {
            "calendar-unchecked": Array [],
          },
          Object {
            "calendar-holiday": Array [],
          },
          Object {
            "calendar-vacation": Array [],
          },
          Object {
            "calendar-otanjoubi": Array [],
          },
          Object {
            "calendar-absence": Array [],
          },
          Object {
            "calendar-future-day": Array [],
          },
        ],
        "inline": true,
        "maxTime": "2018-02-22T12:00:00.000Z",
        "monthsShown": 1,
        "onBlur": [Function],
        "onChange": [MockFunction],
        "onClickOutside": [Function],
        "onFocus": [Function],
        "onKeyDown": [Function],
        "onMonthChange": [Function],
        "onSelect": [Function],
        "onYearChange": [Function],
        "preventOpenOnFocus": false,
        "selected": "2018-02-14T00:00:00.000Z",
        "shouldCloseOnSelect": true,
        "showTimeSelect": false,
        "timeCaption": "Time",
        "timeIntervals": 30,
        "withPortal": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MonthlyCalendar render should show if a day in past is blocked 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MonthlyCalendar
    onDateChange={[MockFunction]}
    selectedDate={"2018-02-21T00:00:00.000Z"}
    timeEntries={
      Array [
        Object {
          "__typename": "TimeEntry",
          "activity": "Code/ Document Reviews",
          "date": "2018-02-11",
          "endBreakTime": "12:35",
          "endTime": "18:28",
          "phase": "Project Phase Name",
          "startBreakTime": "11:35",
          "startTime": "9:28",
          "total": "8:00",
        },
        Object {
          "__typename": "TimeEntry",
          "activity": "Code/ Document Reviews",
          "date": "2018-02-12",
          "endBreakTime": "12:35",
          "endTime": "18:28",
          "phase": "Project Phase Name",
          "startBreakTime": "11:35",
          "startTime": "9:28",
          "total": "8:00",
        },
        Object {
          "__typename": "TimeEntry",
          "activity": "Code/ Document Reviews",
          "date": "2018-02-13",
          "endBreakTime": "12:35",
          "endTime": "18:28",
          "phase": "Project Phase Name",
          "startBreakTime": "11:35",
          "startTime": "9:28",
          "total": "8:00",
        },
        Object {
          "__typename": "TimeEntry",
          "activity": "",
          "date": "2018-02-14",
          "endBreakTime": "",
          "endTime": "",
          "phase": "",
          "startBreakTime": "",
          "startTime": "",
          "total": "",
        },
        Object {
          "__typename": "TimeEntry",
          "activity": "Code/ Document Reviews",
          "date": "2018-02-15",
          "endBreakTime": "12:35",
          "endTime": "18:28",
          "phase": "Project Phase Name",
          "startBreakTime": "11:35",
          "startTime": "9:28",
          "total": "8:00",
        },
        Object {
          "__typename": "TimeEntry",
          "activity": "Code/ Document Reviews",
          "date": "2018-02-16",
          "endBreakTime": "",
          "endTime": "18:28",
          "phase": "Project Phase Name",
          "startBreakTime": "",
          "startTime": "9:28",
          "total": "8:00",
        },
        Object {
          "__typename": "TimeEntry",
          "activity": "Code/ Document Reviews",
          "date": "2018-02-17",
          "endBreakTime": "12:35",
          "endTime": "18:28",
          "phase": "Project Phase Name",
          "startBreakTime": "11:35",
          "startTime": "9:28",
          "total": "8:00",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "allowSameDay": false,
      "dateFormat": "L",
      "dateFormatCalendar": "MMMM YYYY",
      "disabled": false,
      "disabledKeyboardNavigation": false,
      "dropdownMode": "scroll",
      "filterDate": [Function],
      "highlightDates": Array [
        Object {
          "calendar-checked": Array [
            "2018-02-11T02:00:00.000Z",
            "2018-02-12T02:00:00.000Z",
            "2018-02-13T02:00:00.000Z",
            "2018-02-15T02:00:00.000Z",
            "2018-02-16T02:00:00.000Z",
            "2018-02-17T02:00:00.000Z",
          ],
        },
        Object {
          "calendar-unchecked": Array [
            "2018-02-14T02:00:00.000Z",
          ],
        },
        Object {
          "calendar-holiday": Array [],
        },
        Object {
          "calendar-vacation": Array [],
        },
        Object {
          "calendar-otanjoubi": Array [],
        },
        Object {
          "calendar-absence": Array [],
        },
        Object {
          "calendar-future-day": Array [],
        },
      ],
      "inline": true,
      "maxTime": "2018-02-22T12:00:00.000Z",
      "monthsShown": 1,
      "onBlur": [Function],
      "onChange": [MockFunction],
      "onClickOutside": [Function],
      "onFocus": [Function],
      "onKeyDown": [Function],
      "onMonthChange": [Function],
      "onSelect": [Function],
      "onYearChange": [Function],
      "preventOpenOnFocus": false,
      "selected": "2018-02-21T00:00:00.000Z",
      "shouldCloseOnSelect": true,
      "showTimeSelect": false,
      "timeCaption": "Time",
      "timeIntervals": 30,
      "withPortal": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowSameDay": false,
        "dateFormat": "L",
        "dateFormatCalendar": "MMMM YYYY",
        "disabled": false,
        "disabledKeyboardNavigation": false,
        "dropdownMode": "scroll",
        "filterDate": [Function],
        "highlightDates": Array [
          Object {
            "calendar-checked": Array [
              "2018-02-11T02:00:00.000Z",
              "2018-02-12T02:00:00.000Z",
              "2018-02-13T02:00:00.000Z",
              "2018-02-15T02:00:00.000Z",
              "2018-02-16T02:00:00.000Z",
              "2018-02-17T02:00:00.000Z",
            ],
          },
          Object {
            "calendar-unchecked": Array [
              "2018-02-14T02:00:00.000Z",
            ],
          },
          Object {
            "calendar-holiday": Array [],
          },
          Object {
            "calendar-vacation": Array [],
          },
          Object {
            "calendar-otanjoubi": Array [],
          },
          Object {
            "calendar-absence": Array [],
          },
          Object {
            "calendar-future-day": Array [],
          },
        ],
        "inline": true,
        "maxTime": "2018-02-22T12:00:00.000Z",
        "monthsShown": 1,
        "onBlur": [Function],
        "onChange": [MockFunction],
        "onClickOutside": [Function],
        "onFocus": [Function],
        "onKeyDown": [Function],
        "onMonthChange": [Function],
        "onSelect": [Function],
        "onYearChange": [Function],
        "preventOpenOnFocus": false,
        "selected": "2018-02-21T00:00:00.000Z",
        "shouldCloseOnSelect": true,
        "showTimeSelect": false,
        "timeCaption": "Time",
        "timeIntervals": 30,
        "withPortal": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
